### PR TITLE
Added a snippet to show how to read a newline-delimited-json file and store it in a Table

### PR DESCRIPTION
**Pull request by happyhuman**

### CODE REVIEW

File: google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java [hunk 1]
```diff
@@ -378,6 +380,40 @@ public long writeFileToTable(String datasetName, String tableName, Path csvPath)
...
+  public Long writeRemoteFileToTable(String datasetName, String tableName, String sourceUri, List<String> fieldNames)
+      throws InterruptedException {
+    // [START bigquery_load_table_gcs_json]
+    TableId tableId = TableId.of(datasetName, tableName);
+    LoadJobConfiguration configuration = LoadJobConfiguration.builder(tableId, sourceUri)
+        .setFormatOptions(FormatOptions.json())
+        .build();
+    // Table field definition
+    ArrayList<Field> fields = new ArrayList<>();
+    for (String fieldName: fieldNames) {
```
Comment (pongad) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java]: Space before `;`

File: google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java [hunk 2]
```diff
@@ -378,6 +380,40 @@ public long writeFileToTable(String datasetName, String tableName, Path csvPath)
...
+      fields.add(Field.of(fieldName, LegacySQLTypeName.STRING));
+    }
+    // Table schema definition
+    Schema schema = Schema.of(fields.toArray(new Field[fields.size()]));
```
Comment (pongad) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java]: Shouldn't `Schema.of(fields)` work here?

File: google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java [hunk 3]
```diff
@@ -378,6 +380,40 @@ public long writeFileToTable(String datasetName, String tableName, Path csvPath)
...
+    // Create the table
+    StandardTableDefinition tableDefinition = StandardTableDefinition.of(schema);
+    bigquery.create(TableInfo.of(tableId, tableDefinition));
```
Comment (tswast) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java]: You shouldn't have to create the table manually. The load job should create the table for you.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java [hunk 4]
```diff
@@ -378,6 +380,40 @@ public long writeFileToTable(String datasetName, String tableName, Path csvPath)
     // [END writeFileToTable]
   }
 
+  /**
+   * Example of writing a newline-delimited-json file with textual fields from GCS to a table.
+   */
+  // [TARGET writer(WriteChannelConfiguration)]
+  // [VARIABLE "my_dataset_name"]
+  // [VARIABLE "my_table_name"]
+  // [VARIABLE "my_source_uri")]
```
Comment (tswast) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java]: Please use the concrete source URI in the sample itself as is done in the other language samples. `gs://cloud-samples-data/bigquery/us-states/us-states.json`

You should not need to pass this in as an argument.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java [hunk 5]
```diff
@@ -378,6 +380,40 @@ public long writeFileToTable(String datasetName, String tableName, Path csvPath)
...
+  // [VARIABLE "field_names")]
```
Comment (tswast) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java]: Likewise for `field_names`. The schema in the sample should be concrete. That is: don't pass it in, show it being constructed.

Aside: if we were to keep this argument (which we hopefully are not), the variable should be a list of strings, not just a string.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java (hunk 1 above)
Comment (happyhuman) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java]: Done.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java (hunk 2 above)
Comment (happyhuman) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java]: Good point. I had not seen the other available method.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java (hunk 3 above)
Comment (happyhuman) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java]: I am looking into it to see how I can modify the code to let the load job create the table.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java (hunk 4 above)
Comment (happyhuman) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java]: But this was my exact concern in the email. The way the Java project is set up, it is a bit different from the Python code. In the Python code, the entire snippet is written in the unit test. In the Java code, the snippets are written as methods with parameters passed to them and a unit test that send the method the parameters. I had to violate either the "full matching of the Java and Python code" or "how the java project has been set up". Per your suggestion, I violated the former.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java (hunk 5 above)
Comment (happyhuman) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java]: Same as above.

**Comment by happyhuman** [comment]
After talking to Tim, I made some changes. Can you please review the changes? Thanks.

**Comment by tswast** [comment]
Sorry I forgot about this earlier, but there's also a Python script to run which will ensure this sample gets included in the Javadocs for the library. The script is at https://github.com/GoogleCloudPlatform/google-cloud-java/blob/master/utilities/add_snippets_to_file.py

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQuery.java [hunk 6]
```diff
@@ -524,6 +524,19 @@ public int hashCode() {
    * }
    * } </pre>
    *
+   * <p>Example of creating a dataset.
```
Comment (tswast) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQuery.java]: Somehow the script seems to have added this one twice.

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQuery.java [hunk 7]
```diff
@@ -1047,28 +1082,21 @@ TableResult listTableData(
...
    * }
    * }</pre>
    *
    * <p>Example of running a query with query parameters.
-   *
-   * <pre>{@code
-   * String query =
-   *     "SELECT distinct(corpus) FROM `bigquery-public-data.samples.shakespeare` where word_count > ?";
+   * <pre> {@code
+   * String query = "SELECT distinct(corpus) FROM `bigquery-public-data.samples.shakespeare` where;
```
Comment (tswast) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQuery.java]: Something got messed up in this one, too. :-( Half the query string disappeared.

**Pull request merged**